### PR TITLE
Test Front50 use of Google Cloud Storage

### PR DESCRIPTION
**Pull request by ewiseblatt**
@lwander 

This currently tests only Application objects. The critical integration is the same for all the different objects, so I think this is sufficient as a smoke test for the time being.

### CODE REVIEW

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      contract.add_clause(clause)
+
+    # The update timestamp is determined by the server,
+    # and we dont know what that is, so lets ignore it
+    # and assume the unit tests verify it is properly updated.
+    expect = dict(expect)
+    del(expect['updateTs'])
+    f50_builder = st.http_observer.HttpContractBuilder(self.agent)
+    (f50_builder.new_clause_builder('Added Application')
+     .get_url_path(os.path.join('/default/applications/name',
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Where does this string come from?

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    # The createTs we gave ignored.
+    # As before, the name is upper-cased.
+    del expect['updateTs']
+    expect['createTs'] = self.initial_app_spec['createTs']
+    expect['name'] = self.initial_app_spec['name'].upper()
+
+    # TODO(ewiseblatt) 20160524:
+    # Add clauses that observe Front50 to verify the history method works
+    # and that the get method is the current version.
+    num_versions = 2 if self.versioning_enabled else 1
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Very neat, this is exciting.

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  def test_a_create_app(self):
+    self.run_test_case(self.scenario.create_app())
+
+  def test_b_update_app(self):
+    self.run_test_case(self.scenario.update_app())
+
+  def test_z_destroy_app(self):
+    self.run_test_case(self.scenario.destroy_app())
+
+
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Extraneous newlines?

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+     .list_path(
+          os.path.join(self.BUCKET_ROOT, 'applications', self.TEST_APP),
+          with_versions=True)
+     .contains_path_value('name', self.TEST_APP,
+                          min=num_versions, max=num_versions))
+    (builder.new_clause_builder('Updated File Content')
+     .inspect_contents(
+          os.path.join(self.BUCKET_ROOT, 'applications', self.TEST_APP,
+                       'specification.json'))
+#                       'application-metadata.json'))
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Dead comment?

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                                self.TEST_APP))
+     .contains_path_value('', expect))
+    for clause in f50_builder.build().clauses:
+      contract.add_clause(clause)
+
+    path = os.path.join('/default/applications/name/', self.TEST_APP)
+    return st.OperationContract(
+        self.new_post_operation(
+            title='create_app', data=payload, path=path),
+        contract=contract)#builder.build())
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Dead comment?

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+    # file.
+    gcs_builder = gcp.GoogleCloudStorageContractBuilder(self.gcs_observer)
+    (gcs_builder.new_clause_builder('Created Google Cloud Storage File')
+     .list_path(os.path.join(self.BUCKET_ROOT, 'applications'))
+     .contains_path_value('name', self.TEST_APP))
+    (gcs_builder.new_clause_builder('Wrote File Content')
+     .inspect_contents(
+          os.path.join(self.BUCKET_ROOT, 'applications', self.TEST_APP,
+                       'specification.json'))
+     # 'application-metadata.json'))
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Dead comment?

File: spinnaker/spinnaker_system/google_front50_test.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        parser, defaults=defaults)
+
+  def _do_init_bindings(self):
+    """Hook to initialize custom test bindings so journaling is scoped."""
+    config = self.agent.deployed_config
+    enabled = config.get('spinnaker.gcs.enabled', False)
+    if not enabled:
+      raise ValueError('spinnaker.gcs.enabled is not True')
+
+    bucket = config['spinnaker.gcs.bucket']
```
Comment (lwander) [inline spinnaker/spinnaker_system/google_front50_test.py]: Is it not acceptable for people to specify the bucket with a [`gs://` suffix](https://github.com/google/citest/pull/159/files#diff-5a5b021e06e0c7fc8ea1360cfd485c83R58)?

File: spinnaker/spinnaker_system/google_front50_test.py (hunk 1 above)
Comment (ewiseblatt) [inline spinnaker/spinnaker_system/google_front50_test.py]: Added a comment. This is the front50 web server URL to retrieve the application. It isnt directly GCS. I thought I might as well test the web server GET while I was here.

File: spinnaker/spinnaker_system/google_front50_test.py (hunk 7 above)
Comment (ewiseblatt) [inline spinnaker/spinnaker_system/google_front50_test.py]: No. This is the name of the bucket, not a URI.

**Comment by lwander** [comment]
LGTM

**Pull request merged**